src/plugins/data-card/use-data-card-tile-height.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useRef } from "react";
+import { kButtonSpace, kExampleDeckHeight } from "./data-card-types";
+
+interface UseDataCardTileHeightProps {
+  tileElt: HTMLElement | null;
+  height: number;
+  currEditAttrId: string;
+  modelId: string;
+  documentId: string | undefined;
+  readOnly: boolean;
+  onRequestRowHeight: (id: string, height: number) => void;
+  attrCount: number;
+}
+
+export const useDataCardTileHeight = ({
+  tileElt,
+  height,
+  currEditAttrId,
+  modelId,
+  documentId,
+  readOnly,
+  onRequestRowHeight,
+  attrCount
+}: UseDataCardTileHeightProps) => {
+
+  const attrCtRef = useRef(attrCount);
+
+  useEffect(() => {
+    if (!tileElt) return;
+    const uiHeight = tileElt.querySelector(".data-card-container")?.scrollHeight || 0;
+    const spaceLeft = height ? height - uiHeight : 0;
+    const requiringHeightChange = attrCtRef.current < attrCount || spaceLeft < kButtonSpace;
+    const loadingInContent = !documentId && readOnly;
+
+    if (loadingInContent) {
+      onRequestRowHeight(modelId, Math.max(uiHeight, kExampleDeckHeight));
+    }
+
+    if (requiringHeightChange) {
```

Comment:
I'd think you'd want to make this an `else if`. Otherwise even if `loadingInContent` is true, `requiringHeightChange` might be true, and then the height would be updated to include the `kButtonSpace`.

Also if the document is readOnly it seems like `requiringHeightChange` could be true. So I'd guess you'd want to update `requiringHeightChange`'s definition. To be something like:
`const requiringHeightChange = !readOnly && (attrCtRef.current < attrCount || spaceLeft < kButtonSpace);`